node/network/collator-protocol/src/lib.rs

```diff
@@ -44,6 +46,15 @@ mod validator_side;
 
 const LOG_TARGET: &'static str = "parachain::collator-protocol";
 
+/// The maximum depth a candidate can occupy for any relay parent.
+/// 'depth' is defined as the amount of blocks between the para
+/// head in a relay-chain block's state and a candidate with a
+/// particular relay-parent.
+///
+/// This value is only used for limiting the number of candidates
+/// we accept and distribute per relay parent.
+const MAX_CANDIDATE_DEPTH: usize = 4;
```

Comment:
At some point we'd like for this to be configurable by the relay-chain, probably as part of session-info. If it is constant, it should be higher up so it can be reused within multiple subsystems. e.g. in node-primitives.